pkg/azurecluster/validate_update.go

```diff
@@ -67,18 +68,38 @@ func (a *UpdateValidator) Validate(ctx context.Context, request *v1beta1.Admissi
 		return microerror.Mask(err)
 	}
 
+	return a.validateRelease(ctx, azureClusterOldCR, azureClusterNewCR)
+}
+
+func (a *UpdateValidator) Log(keyVals ...interface{}) {
+	a.logger.Log(keyVals...)
+}
+
+func (a *UpdateValidator) validateRelease(ctx context.Context, azureClusterOldCR *capzv1alpha3.AzureCluster, azureClusterNewCR *capzv1alpha3.AzureCluster) error {
 	oldClusterVersion, err := semverhelper.GetSemverFromLabels(azureClusterOldCR.Labels)
 	if err != nil {
-		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from AzureConfig (before edit)")
+		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from the AzureCluster being updated")
 	}
 	newClusterVersion, err := semverhelper.GetSemverFromLabels(azureClusterNewCR.Labels)
 	if err != nil {
-		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from AzureConfig (after edit)")
+		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from applied AzureCluster")
 	}
 
-	return releaseversion.Validate(ctx, a.ctrlClient, oldClusterVersion, newClusterVersion)
-}
+	if !newClusterVersion.Equals(oldClusterVersion) {
+		cluster, err := capiutil.GetOwnerCluster(ctx, a.ctrlClient, azureClusterNewCR.ObjectMeta)
+		if err != nil {
+			return microerror.Mask(err)
+		}
 
-func (a *UpdateValidator) Log(keyVals ...interface{}) {
-	a.logger.Log(keyVals...)
+		clusterCRReleaseVersion, err := semverhelper.GetSemverFromLabels(cluster.Labels)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if !newClusterVersion.Equals(clusterCRReleaseVersion) {
+			return microerror.Maskf(errors.InvalidOperationError, "AzureCluster release version must be set to the same release version as Cluster CR release version label")
+		}
```

Comment:
@fiunchinho we blocked azure-operator from upgrading `AzureCluster` :see_no_evil: I changed this to a simple check that the new `AzureCluster` release simply matches `Cluster` release.

Reply:
Ok, that makes sense.

Reply:
Works now, `AzureCluster` release label is updated, upgrade is in progress.